pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -669,6 +681,18 @@ func (c *Controller) getForeignServiceInstancesByPort(svc *model.Service, reqSvc
 		return nil
 	}
 
+	// Now get the target Port for this service port
+	k8sService := item.(*v1.Service)
+	targetPort := reqSvcPort
+	for _, p := range k8sService.Spec.Ports {
+		if p.Name == servicePort.Name {
+			if p.TargetPort.Type == intstr.Int && p.TargetPort.IntVal > 0 {
```

Comment:
is there not a way to support `targetPort: name`? That would allow people to have a different port on each VM which was one of @costinm concerns

Reply:
if we shove the endpoint port map into each endpoint object and pass it around, we could. Not sure it is worth the hassle especially given the k8s issue I pointed out, where the response was "it works properly for numeric ports". How about we see the demand for this before taking up this aspect?

Reply:
and I tried doing that exactly, except I hit this bug https://github.com/kubernetes/kubernetes/issues/54213 .. which led me to believe that doing this endpoint port map will result undefined behavior if the container ports are not unique. I would much rather say target port by name is not supported than try to debug this behavior in the wild.